sqlglot/optimizer/qualify_columns.py

```diff
@@ -410,7 +411,9 @@ def _expand_stars(
             else:
                 return
 
-    scope.expression.set("expressions", new_selections)
+    # Ensures we don't overwrite the initial selections with an empty list
+    if new_selections:
+        scope.expression.set("expressions", new_selections)
```

Comment:
This isn't currently necessary for the tests to pass, but it was when I wrote it because I ended up with a query like `SELECT FROM ...` due to `new_selections` being empty.

We can remove it or keep it, but imo it's worth having this as a sanity check since it's cheap.

Reply:
i've seen this before too

Reply:
it'd be good to add a test for it if possible

Reply:
yep I'll try to come up with a test case that covers it

Reply:
added a test, see https://github.com/tobymao/sqlglot/pull/1916/commits/22803c03dc503f4c7ea7bdd9322f59ce9eea25e7